Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -696,6 +697,7 @@ public static Branch getAutoInstance(@NonNull Context context) {
         customReferrableSettings_ = CUSTOM_REFERRABLE_SETTINGS.USE_DEFAULT;
         boolean isTest = BranchUtil.checkTestMode(context);
         getBranchInstance(context, !isTest, null);
+        new BranchPreinstall().getPreinstallSystemData(context);
```

Comment:
Why not pass in the newly created `Branch` object?   Instead of having the class immediately call `Branch::getInstance()` 3 times

I suspect that if Branch is null for some reason that we are not checking.   May not be possible though.

Reply:
Done, checking for null also.